### DISCUSSION TITLE
Ma chi ve lo ha detto che non si possono inserire le accentate?

### DISCUSSION THREAD

**Discussion by crazyLillo** [tesivo/tesivo_s_VTMB_italian_translation_for_Unofficial_Patch#64] [Generale]
Ecco fatto! tutte le accentate del mondo. 
Ancora non controllo per bene, ma mi sa che qualcosina che non va ancora c'è.
Il motore di dialogo ha dei bug noti, irrisolvibili al momento...

![jackaccentato](https://user-images.githubusercontent.com/96891196/154509913-12939c3f-d190-4857-86e0-d7a873d3981a.jpg)

**Comment by NarakuITA** [comment]
Che bug per l'esattezza?

**Reply by crazyLillo** [reply]
Allora, non conosco le specifiche dei bugs in questione, ma ho letto (in Inglese) da qualche parte nelle mie vaste esplorazioni, 
che il motore di dialogo ha dei piccoli bugs fin dall'inizio, che non sono mai stati risolti integralmente. 
Ma sembra che in Inglese il gioco non dia particolari problemi, (forse perchè hanno pensato solo a loro...booo.)

Sono dei bugs che non si manifestano sempre e comunque, ma in modo random che forse dipendono anche da varie configurazioni hardware dei pc.
A quanto pare, sono gli utenti di altre lingue ad avere la possibilità che quei problemi si manifestino.

All'inizio dietro vostri consigli, usavo anchio notepad++, e mi ero adeguato al fatto che le accentate andavano simulate con l'apostrofo.

Ma poi come sai, ho scoperto quel programmino per localizzare, all'interno dell'SDK.

Lo hanno fatto dei ragazzi della cecoslovacchia, e ripensandoci l'altro giorno, mi sono soffermato sul fatto, 
che la loro lingua è strapiena di accenti molto più della nostra.

Quella lingua senza accenti sembra un linguaggio per celebrolesi, quindi mi chiedevo che cosa abbiano fatto per ovviare a un apparente insormontabile problema.
Ti ricordo che loro non possono simulare tutti quegli accenti solo con gli apostrofi...

Allora mi viene un dubbio atroce, faccio la prova!
(che non avevo fatto all'inizio, perchè davo ormai per scontato che le accentate, bla, bla ,bla...)

Come vedi il programmino le accentate le inserisce eccome!
...e bravi i ragazzi della cecoslovacchia!

Io non ho meriti di nessun tipo, solo curiosità di sperimentare...

Adesso il problema è un altro,

Il problema sono i file LIP (i sottotitoli che appaiono in alto a sinistra).

Lì veramente non accetta le accentate, anche con l'utility apposita, creata da altri.

mi sa che lì gli apostrofi saranno necessari

devo fare delle prove, ma indagherò anche su questo...

**Reply by tesivo** [reply]
Interessante indagine. Potrebbe riguardare la tabella dei caratteri a cui fa riferimento il sistema operativo sottostante, o il modo in cui vengono gestiti i caratteri dal processore. 

Colgo l'occasione per dirvi che sarò ancora piuttosto assente per una 30 di giorni, poi tornerò, e se tale mistero ci sarà ancora sarò ben lieto di aiutarvi a fare un po' di luce.

**Reply by NarakuITA** [reply]
Se mi passi il link del forum dove hai scaricato posso provare a parlare con gli autori. O magari puoi provare tu. Magari ci daranno delle dritte.

**Reply by crazyLillo** [reply]
tesivo, non credo si tratti di quello, ma non scarto a priori alcuna possibilità.

NarakuIta, 
se ti stai rivolgendo a me,
Scusa, mica ho capito bene cosa intendi per link? (di quale link parli?)
Di quale forum parli?
La cosa "scaricata" che sarebbe?

**Comment by NarakuITA** [comment]
Del programma per localizzare, all'interno dell'SDK.

**Reply by crazyLillo** [reply]
Penso che ci sia stato un equivoco!

non c'è nessun link e nessun forum di mezzo

Allora, se intendi i programmini per localizzare i dialoghi e i sottotitoli, quelli te li trovi solo se quando installi
l'unofficial patch, metti la spunta per installare i file dell'SDK (che sarebbe la "Patch_Extras") da non confondere con la "Plus Patch".

Se non attivi nulla, nulla succede!, non ti crea fastidio alcuno, avere quella cartella, a meno che non vai a smanettare senza cognizione di causa...

In pratica ti ritrovi nella cartella di installazione del gioco la cartella "Patch_Extras" con dentro i file dell'SDK.
L'SDK, è un file exe che una volta attivato, da lì puoi attivare anche tutti gli altri file exe di tutti gli strumenti all'interno.

All'interno della cartella "Patch_Extras" ci sono molte cartelle diverse, e molti file exe dappertutto.
Ma non c'è bisogno di toccare nulla, a meno che non vuoi creare un mod da zero.
C'è una cartella al'interno di altre cartelle (bisogna proprio andarla a cercare), la cartella da cercare si chiama "Tools".
All'interno di questa cartella, tante altre cartelle (una 20 ina e più) dove all'interno di ciascuna di esse vi è un programma, che svolge qualcosa di specifico.
Dipende quello che vuoi fare ed ottenere.

Questi programmini possono essere attivati anche singolarmente senza stare a scomodare il programma principale (che non ci serve)

Detto questo, io ho trovato lì (in cartelle diverse), quei programmi per la localizzazione e sottotitoli, di cui ho creato un link sul desktop per attivarli.
 occhio, che ad ogni unofficial patch, l'SDK viene aggiornato anche lui che adesso è alla versione 2.0 (con l'unofficial patch 11.1).
Qualche tool è rimasto lo stesso, altri aggiornati, altri eliminati.

Comunque ci sono i nick degli autori, su quei tools, ma dove li vai a pescare? - sono ancora attivi dopo anni?) - risponderanno?

Andarli a cercare significa per me una perdita di tempo.

**Reply by crazyLillo** [reply]
--------------------------------------------------------------------





---------------------------------------------------------------------

Adesso mi sono accorto di un ulteriore problema nuovo

Riguarda i file LIP dei sottotitoli e l'inserimento delle accentate (non sono come i file DLG di cui il problema ho già risolto).

Mi sto accorgendo che qualcuno ha fatto un insalata mista di questi file LIP, per quante volte sono stati manipolati.
i file dovrebbero essere dello stesso identico tipo, e all'apparenza lo sono, ma guardando la codifiche mi viene da storcere il naso.

Alcuni file sono , Windows (cr lf) Utf-8. e altri, Windows (cr lf) Ansi.

Alcuni, Unix (lf) Ansi, e altri  Unix (lf) Utf-8.

Non me lo spiego, ma è alquanto strano... (anche i file originali sono così...)

Ad ogni modo, se utilizzi l'utility apposita per inserire i sottotitoli, le accentate non te le accetta,
anche se le inserisci, non vedi caratteri strani, ma la lattera accentata inserita viene semplicemente sostituita da una non accentata.

se utilizzi notepad++, accade qualcosa di diverso, (sto ancora indagando), ma succede un casino per colpa di tutte quelle codifiche diverse.

**Reply by NarakuITA** [reply]
Dovrei chiedere a chi a fatto la patch non ufficiale...

**Reply by NarakuITA** [reply]
Magari sanno dirmi qualcosa, un suggerimento, non so. Cosa dici?

**Reply by crazyLillo** [reply]
*************************************************************************

Allora, per quanto riguarda l'unofficial patch, (wesp) , l'autore, si fa aiutare da una comunità intera ancora molto attiva, e accetta migliorie di qualunque genere.

Ti ricordi che postai dei file aggiustati che si erano rivelati corrotti? Appunto perchè manipolati in modo strano o senza cognizione di causa.

Ho il sospetto che queste corruzioni, avvengono anche se con i file ci lavori con Linux...e dipende anche da che programma usi. 
(ma non posso averne conferma, e solo un sospetto e se monto la mia "Black Hat Linux" (che non uso da anni, perchè ho chiuso con quel mondo) solo per verificare,
non mi vedete più...

Poi c'è da dire che quelli dell'unofficial patch, non sono i file originali veri e propri, ma manipolati appunto per creare la patch.

i veri originali sono all'interno di file molto compressi nella cartella "Vampire" che è fuori dalla cartella dell'unofficial patch.
e se il gioco lo hai installato, li puoi vedere anche tu.

Per scompattarli ci vogliono ore (così ho letto...) oltre ad usare per forza L'SDK per farlo. (non puoi usare altro per vari motivi...)

Comunque ho letto anche che quei file LIP a detta di wesp, dovrebbero essere (TrueUnicode UTF-16.)
Forse per far la patch li hanno convertiti in qualche modo...

Non so quanto valga la pena contattare wesp, se la domanda non gli interessa quello non ti risponde neanche...o forse si...

se mi iscrivo in quei forum dove sta lui, poi non lo so se ne vengo fuori o mi ci perdo ( ho una traduzione da fare...)


-------------------------------------------------

Adesso l'altra questione

Il programma che serve per i sottotitoli è questo. (usato anche per Half Life)

LipSync/VCD Editor (Valve's FacePoser modification, extended by Psycho-A)

Questo lo trovi solo nel sdk fornito con la patch 11.1, che ha sostituito la vecchia utility nelle patch precedenti.
L'autore Psycho-A, lo ha modificato e migliorato.

-------------------------------------------------------------------------------------------------------------

Spulciando qui e li ho trovato anche il suo indirizzo email :

You may also email me: psycho-a@rambler.ru, psycho-a@yandex.com.

Non mi azzardo nemmeno da lontano a contattare questo Russo. (non li digerisco, cose mie...niente di politico...solo cose del passato).

**Reply by crazyLillo** [reply]
------------------------------------------------------------------------

Collegandomi al discorso precedente, dovrebbe esserci anche l'email di wesp in qualche file dell'sdk, solo che adesso non la trovo più, ma sono sicuro che da qualche parte in quei file c'è.

---------------------------------------------------------------------------

Comunque per fargli accettare le accentate (nei file LIP) è solo questione di azzeccare la codifica giusta (per noi) dei file.
la brutta notizia è che in questo modo l'utility (lip-editor) non la possiamo usare finchè qualcuno non la modifica per i nostri scopi, o almeno questo è quello che credo.

La codifica giusta la si può inserire solo con notepad++, ritornando a contare il numero dei caratteri in modo manuale (che palle...).

Sto facendo delle prove, e sembra che qualcosa si sta muovendo, nell'immagine potete vedere che le accentate ci sono!
Ma non riesco a capire perchè i sottotitoli adesso scorrono velocissimi che quasi non fai in tempo a leggerli. boooo
continuerò a provare finche non imbrocco la via giusta...


![prince](https://user-images.githubusercontent.com/96891196/154841331-14d34e13-3d2f-4166-a431-9f2fd73d13f6.jpg)

**Reply by NarakuITA** [reply]
Pian piano, stai risolvendo l'enigma. Secondo me a questo punto, ti conviene finire la traduzione. Poi come diceva Tesivo si passa a ottimizzarla, li poi uno si concentra a risolvere questi "problemi".

**Reply by NarakuITA** [reply]
Anche perché ci sono dei bug che la traduzione ha fatto emergere. E non so o non ricordo se magari, visto che avevi revisionato tutto il tutorial, avevi corretto. C'è un punto dove non è possibile proseguire. Quando devi imparare a scassinare la porta.

**Reply by crazyLillo** [reply]
--------------------------------------------------------------------------------


si hai ragione, mi sto fossilizzando troppo su questo problema delle accentate nei file Lip.
ma mi sta facendo un pò sclerare per motivi presumibilmente occulti (la presunta maledizione del gioco...).

Provo una cosa e sembra che va bene, esco dal gioco, lo riprovo e non va più bene......BHAAAAA...

al massimo per i sottotitoli si usano gli apostrofi e pace. (vedremo).

C'è da dire anche che ci sono un interminabile sfilza di messaggi da correggere che sono con gli apostrofi, parlo dei messaggi a  video che vengono quando stai caricando etc...
Tipo "L'abbraccio e' la trasformazione di un mortale in un vampiro...", " il mondo di tenebra e bla bla bla..". (hai capito di che nessaggi parlo?), un lavoraccio...

------------------------------------------------------------------

si le traduzioni introducono bugs a palate per il fatto che non sono dei semplici file di testo, dove anche un semplice spazio in più nel posto sbagliato può significare un blocco.

Noo, il mio tutorial fila liscio che è una bellezza. l'ho rigiocato un interminabile numero di volte per vedere la mia traduzione in diretta e non si è mai e poi mai bloccato.

Purtroppo il file del tutorial è tutto da rivedere per inserire tutti gli accenti nei posti giusti.
io ho fatto solo delle prove sulle prime frasi per vedere se si vedevano gli accenti, ma il resto bisogna correggerlo tutto.

Al momento sto ritraducendo da zero il file beckett.dlg (uno fra i più lunghi e sono all'80%)
Mi son ricopiato i testi su blocco note per lavorarci senza paura di introdurre bugs e senza distrazioni,
poi copierò il tutto con il programmino apposito, così come ho fatto per il tutorial.
